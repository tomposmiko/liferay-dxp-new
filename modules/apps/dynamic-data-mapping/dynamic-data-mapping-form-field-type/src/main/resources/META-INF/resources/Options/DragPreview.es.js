/**
 * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
 *
 * This library is free software; you can redistribute it and/or modify it under
 * the terms of the GNU Lesser General Public License as published by the Free
 * Software Foundation; either version 2.1 of the License, or (at your option)
 * any later version.
 *
 * This library is distributed in the hope that it will be useful, but WITHOUT
 * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
 * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
 * details.
 */

import React, {useRef} from 'react';
import {useDragLayer} from 'react-dnd';
import ReactDOM from 'react-dom';

import {OPTIONS_TYPES} from './DnD.es';

const layerStyles = {
	height: '100%',
	left: 0,
	pointerEvents: 'none',
	position: 'fixed',
	top: 0,
	width: '100%',
	zIndex: 100,
};

const getItemStyles = (currentOffset, ref) => {
	if (!currentOffset || !ref.current) {
		return {
			display: 'none',
		};
	}

	const {x, y} = currentOffset;
	const transform = `translate(${x}px, ${y}px)`;

	return {
		WebkitTransform: transform,
		transform,
	};
};

export default function DragPreview({children, component: Component}) {
	const ref = useRef();

	const {currentOffset, isDragging, item} = useDragLayer((monitor) => ({
		currentOffset: monitor.getClientOffset(),
		isDragging: monitor.isDragging(),
		item: monitor.getItem(),
	}));

	if (!isDragging || (isDragging && item.type !== OPTIONS_TYPES.OPTION)) {
		return null;
	}

	return ReactDOM.createPortal(
		<div style={layerStyles}>
			<Component
				{...item.option}
				className="dragging"
				ref={ref}
				style={getItemStyles(currentOffset, ref)}
			>
				{children({index: item.position, option: item.option})}
			</Component>
		</div>,
		document.body
	);
}
