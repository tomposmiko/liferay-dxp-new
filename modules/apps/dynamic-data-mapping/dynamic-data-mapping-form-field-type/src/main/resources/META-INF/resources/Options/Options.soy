{namespace Options}

{template .render}
	{@param name: string}
	{@param spritemap: string}
	{@param? _handleOptionDeleted: any}
	{@param? _handleOptionFocused: any}
	{@param? _handleOptionLabelEdited: any}
	{@param? _handleOptionValueEdited: any}
	{@param? defaultLanguageId: string}
	{@param? displayErrors: bool}
	{@param? editingLanguageId: string}
	{@param? id: string}
	{@param? items: list<[
		disabled: bool,
		generateKeyword: bool,
		label: string,
		value: string
	]>}
	{@param? label: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? tip: string}
	{@param? visible: bool}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{if $items}
				<div class="ddm-field-options-container">
					{foreach $option in $items}
						<div class="ddm-options-target" data-index="{index($option)}" key="target_{index($option)}"></div>

						<div class="ddm-field-options" data-index="{index($option)}" key="{index($option)}">
							<span class="ddm-options-drag {$readOnly ? 'disabled': ''}">
								<svg
									aria-hidden="true"
									class="lexicon-icon lexicon-icon-drag"
								>
									<use xlink:href="{$spritemap}#drag" />
								</svg>
							</span>

							<div class="ddm-option-entry">
								{call KeyValue.render}
									{param events: [
										'fieldEdited': $_handleOptionLabelEdited,
										'fieldFocused' : $_handleOptionFocused,
										'fieldKeywordEdited': $_handleOptionValueEdited
									] /}
									{param generateKeyword: $option.generateKeyword /}
									{param key: 'option' + index($option) /}
									{param keyword: $option.value /}
									{param keywordReadOnly: $defaultLanguageId != $editingLanguageId /}
									{param index: index($option) /}
									{param name: 'option' + index($option) /}
									{param placeholder: $placeholder /}
									{param readOnly: $option.disabled /}
									{param ref: 'option' + index($option) /}
									{param required: $required /}
									{param showLabel: false /}
									{param spritemap: $spritemap /}
									{param value: $option.label /}
								{/call}

								{if not isLast($option) and not $readOnly}
									<button class="close close-modal" data-onclick="{$_handleOptionDeleted}" type="button">
										<svg
											aria-hidden="true"
											class="lexicon-icon lexicon-icon-times"
										>
											<use xlink:href="{$spritemap}#times" />
										</svg>
									</button>
								{/if}
							</div>
						</div>
					{/foreach}
				</div>
			{/if}
		{/param}
		{param displayErrors: $displayErrors /}
		{param id: $id /}
		{param label: $label /}
		{param name: $name /}
		{param repeatable: $repeatable /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
		{param visible: $visible /}
	{/call}
{/template}