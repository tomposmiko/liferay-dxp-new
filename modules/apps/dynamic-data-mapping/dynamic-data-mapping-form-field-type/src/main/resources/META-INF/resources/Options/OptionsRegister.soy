{namespace OptionsRegister}

{deltemplate PageRenderer.RegisterFieldType variant="'options'"}
	{call Options.render data="all" /}
{/deltemplate}