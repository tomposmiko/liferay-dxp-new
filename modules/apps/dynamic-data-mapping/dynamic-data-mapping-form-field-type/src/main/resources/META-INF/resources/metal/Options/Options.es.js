import '../FieldBase/FieldBase.es';
import '../KeyValue/KeyValue.es';
import './OptionsRegister.soy.js';

import Component from 'metal-component';
import dom from 'metal-dom';
import Soy from 'metal-soy';
import templates from './Options.soy.js';
import {Config} from 'metal-state';
import {Drag, DragDrop} from 'metal-drag-drop';
import {
	normalizeFieldName
} from 'dynamic-data-mapping-form-builder/js/components/LayoutProvider/util/fields.es';

/**
 * Options.
 * @extends Component
 */

class Options extends Component {
	attached() {
		let defaultOption = false;
		const defaultOptionLabel = Liferay.Language.get('option').toLowerCase();

		const options = this.getCurrentLocaleValue();

		if ((options.length == 1) && (options[0].label.toLowerCase() == defaultOptionLabel)) {
			defaultOption = true;
		}

		this.setState(
			{
				defaultOption
			}
		);

		this._createDragDrop();
	}

	deleteOption(deletedIndex) {
		const {editingLanguageId} = this;
		let {value} = this;

		for (const languageId in value) {
			value = {
				...value,
				[languageId]: value[languageId].filter(
					(option, currentIndex) => currentIndex !== deletedIndex
				)
			};
		}

		this.setState(
			{
				items: this.getItems(value[editingLanguageId])
			},
			() => this._handleFieldEdited({}, value)
		);
	}

	disposeDragAndDrop() {
		if (this._dragAndDrop) {
			this._dragAndDrop.dispose();
		}
	}

	disposeInternal() {
		super.disposeInternal();

		this.disposeDragAndDrop();
	}

	findOptionByValue(options, name, limit = options.length) {
		return options.find(
			({value}, index) => {
				return index < limit && value === name;
			}
		);
	}

	getCurrentLocaleValue() {
		const {defaultLanguageId, editingLanguageId} = this;
		let value = [];

		if (this.value && this.value[editingLanguageId]) {
			value = this.value[editingLanguageId];
		}
		else if (this.value && this.value[defaultLanguageId]) {
			value = this.value[defaultLanguageId];
		}

		return value;
	}

	getFieldIndex(element) {
		return parseInt(dom.closest(element, '.ddm-field-options').dataset.index, 10);
	}

	getItems(options = []) {
		const {defaultLanguageId, editingLanguageId} = this;
		const items = options.filter(({value}) => !!value);

		if (defaultLanguageId === editingLanguageId) {
			items.push(
				{
					label: '',
					value: ''
				}
			);
		}

		return items.map(
			option => {
				return {
					...option,
					generateKeyword: this.shouldGenerateOptionValue(option)
				};
			}
		);
	}

	moveOption(sourceIndex, targetIndex) {
		const {editingLanguageId} = this;
		let {value} = this;

		for (const languageId in value) {
			const options = [...value[languageId]];

			if (sourceIndex < options.length) {
				options.splice(
					targetIndex,
					0,
					{
						...options[sourceIndex]
					}
				);

				value = {
					...value,
					[languageId]: options.filter(
						(option, index) => {
							return sourceIndex > targetIndex ?
								index != (sourceIndex + 1) :
								index != sourceIndex;
						}
					)
				};
			}
		}

		this.setState(
			{
				items: this.getItems(value[editingLanguageId])
			},
			() => this._handleFieldEdited({}, value)
		);
	}

	normalizeOption(options, option, force) {
		const {label, value} = option;
		const desiredValue = value || label || (force ? Liferay.Language.get('option') : '');
		let normalizedValue = desiredValue;

		if (this.shouldGenerateOptionValue(option)) {
			let counter = 0;
			const optionIndex = options.indexOf(option);

			do {
				if (counter > 0) {
					normalizedValue = desiredValue + counter;
				}

				counter++;
			} while (this.findOptionByValue(options, normalizedValue, optionIndex));

			normalizedValue = normalizeFieldName(normalizedValue);
		}

		return {
			...option,
			value: normalizedValue
		};
	}

	normalizeOptions(options, force) {
		return options.map(option => this.normalizeOption(options, option, force));
	}

	normalizeValue(value, force = false) {
		const newValue = {};

		for (const locale in value) {
			const options = value[locale] || [];

			newValue[locale] = this.normalizeOptions(options, force);
		}

		return newValue;
	}

	shouldGenerateOptionValue(option) {
		const {defaultLanguageId, editingLanguageId} = this;

		return (
			defaultLanguageId === editingLanguageId &&
			(
				option.value === '' ||
				(new RegExp(`^${normalizeFieldName(option.label)}\\d*$`)).test(option.value)
			)
		);
	}

	syncEditingLanguageId(editingLanguageId) {
		const {defaultLanguageId} = this;

		if (
			defaultLanguageId !== editingLanguageId &&
			!this.value[editingLanguageId]
		) {
			this.setState(
				{
					value: {
						...this.value,
						[editingLanguageId]: this.value[defaultLanguageId]
							.filter(({value}) => !!value)
					}
				},
				() => this._handleFieldEdited({}, this.value)
			);
		}
	}

	syncValue() {
		this.setState(
			{
				items: this.getItems(this.getCurrentLocaleValue())
			}
		);
	}

	_createDragDrop() {
		this._dragAndDrop = new DragDrop(
			{
				container: this.element,
				dragPlaceholder: Drag.Placeholder.CLONE,
				handles: '.ddm-options-drag:not(.disabled)',
				sources: '.ddm-field-options',
				targets: '.ddm-options-target',
				useShim: false
			}
		);

		this._dragAndDrop.on(DragDrop.Events.END, this._handleDragDropEvent.bind(this));
		this._dragAndDrop.on(DragDrop.Events.DRAG, this._handleDragEvent.bind(this));
	}

	_getOptionIndex({name}) {
		return parseInt(name.replace('option', ''), 10);
	}

	_handleDragDropEvent({target, source}) {
		const lastSource = document.querySelector('.ddm-source-dragging');
		const sourceIndex = parseInt(source.dataset.index, 10);

		if (lastSource) {
			lastSource.classList.remove('ddm-source-dragging');
		}

		if (target) {
			const targetIndex = parseInt(target.dataset.index, 10);

			this.moveOption(sourceIndex, targetIndex);
		}
	}

	_handleDragEvent({source}) {
		source.classList.add('ddm-source-dragging');
	}

	_handleFieldEdited({originalEvent}, value) {
		this.emit(
			'fieldEdited',
			{
				fieldInstance: this,
				originalEvent,
				value
			}
		);
	}

	_handleOptionBlurred(event) {
		const {value} = this;
		const normalizedValue = this.normalizeValue(value, true);

		this._handleFieldEdited(event, normalizedValue);
	}

	_handleOptionDeleted(event) {
		const {delegateTarget} = event;
		const deletedIndex = this.getFieldIndex(delegateTarget);

		this.deleteOption(deletedIndex);
	}

	_handleOptionEdited(event, property) {
		const {defaultLanguageId, editingLanguageId} = this;
		const {fieldInstance, value} = event;
		let options = this.getCurrentLocaleValue();

		const optionExists = options.some(
			(option, index) => {
				return index === this._getOptionIndex(fieldInstance);
			}
		);

		if (optionExists) {
			options = options.map(
				(option, index) => {
					return index === this._getOptionIndex(fieldInstance) ? (
						{
							...option,
							edited: property === 'label',
							[property]: value
						}
					) : option;
				}
			);
		}
		else {
			options = [
				...options,
				{
					label: property === 'label' ? value : '',
					value: property === 'value' ? value : ''
				}
			];
		}

		if (property === 'label') {
			options = this.normalizeOptions(options);
		}

		let newValue = {
			...this.value,
			[editingLanguageId]: options
		};

		if (defaultLanguageId === editingLanguageId) {
			const generateLabels = (languageId, options) => {
				return options.map(
					({label, value}, index) => {
						const option = newValue[languageId][index];

						if (option && option.edited) {
							label = option.label;
						}

						return {
							edited: option && option.edited,
							label,
							value
						};
					}
				);
			};

			for (const languageId in this.value) {
				if (defaultLanguageId === languageId) {
					continue;
				}

				newValue = {
					...newValue,
					[languageId]: generateLabels(languageId, options)
				};
			}
		}

		this.setState(
			{
				value: newValue
			},
			() => this._handleFieldEdited(event, newValue)
		);
	}

	_handleOptionFocused({originalEvent: {target}}) {
		if (this.defaultOption) {
			target.value = '';

			this.setState(
				{
					defaultOption: false
				}
			);
		}
	}

	_handleOptionLabelEdited(event) {
		this._handleOptionEdited(event, 'label');
	}

	_handleOptionValueEdited(event) {
		this._handleOptionEdited(event, 'value');
	}

	_internalItemsValueFn() {
		const options = this.getCurrentLocaleValue();

		return this.getItems(options || []);
	}

	_setValue(value = {}) {
		const {defaultLanguageId} = this;
		const formattedValue = {...value};

		for (const languageId in value) {
			if (defaultLanguageId !== languageId) {
				formattedValue[languageId] = formattedValue[languageId]
					.filter(({value}) => !!value);
			}
		}

		return formattedValue;
	}
}

Options.STATE = {

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?string}
	 */

	defaultLanguageId: Config.string(),

	/**
	 * @default false
	 * @instance
	 * @memberof Options
	 * @type {?bool}
	 */

	defaultOption: Config.bool().internal().value(false),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?string}
	 */

	editingLanguageId: Config.string(),

	/**
	 * @default 'boolean'
	 * @instance
	 * @memberof Options
	 * @type {?(boolean|undefined)}
	 */

	evaluable: Config.bool().value(false),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?string}
	 */

	fieldName: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?(string|undefined)}
	 */

	items: Config.arrayOf(
		Config.shapeOf(
			{
				disabled: Config.bool().value(false),
				label: Config.string(),
				name: Config.string(),
				value: Config.string()
			}
		)
	).internal().valueFn('_internalItemsValueFn'),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?(string|undefined)}
	 */

	label: Config.string(),

	/**
	 * @default enter-an-option
	 * @instance
	 * @memberof Options
	 * @type {?(string)}
	 */

	placeholder: Config.string().value(Liferay.Language.get('enter-an-option')),

	/**
	 * @default false
	 * @instance
	 * @memberof Options
	 * @type {?bool}
	 */

	readOnly: Config.bool().value(false),

	/**
	 * @default undefined
	 * @instance
	 * @memberof FieldBase
	 * @type {?(bool|undefined)}
	 */

	repeatable: Config.bool(),

	/**
	 * @default false
	 * @instance
	 * @memberof Options
	 * @type {?bool}
	 */

	required: Config.bool().value(false),

	/**
	 * @default true
	 * @instance
	 * @memberof Options
	 * @type {?bool}
	 */

	showLabel: Config.bool().value(true),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?(string|undefined)}
	 */

	spritemap: Config.string(),

	/**
	 * @default undefined
	 * @instance
	 * @memberof Options
	 * @type {?(string|undefined)}
	 */

	tip: Config.string(),

	/**
	 * @default options
	 * @instance
	 * @memberof Options
	 * @type {?(string|undefined)}
	 */

	type: Config.string().value('options'),

	/**
	 * @default {}
	 * @instance
	 * @memberof Options
	 * @type {?string}
	 */

	value: Config.object().setter('_setValue').value({})
};

Soy.register(Options, templates);

export default Options;