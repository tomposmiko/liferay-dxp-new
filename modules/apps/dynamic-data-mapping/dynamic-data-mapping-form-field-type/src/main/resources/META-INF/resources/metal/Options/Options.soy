{namespace Options}

{template .render}
	{@param spritemap: string}
	{@param? _handleOptionBlurred: any}
	{@param? _handleOptionDeleted: any}
	{@param? _handleOptionFocused: any}
	{@param? _handleOptionLabelEdited: any}
	{@param? _handleOptionValueEdited: any}
	{@param? defaultLanguageId: string}
	{@param? editingLanguageId: string}
	{@param? id: string}
	{@param? items: list<[
		disabled: bool,
		generateKeyword: bool,
		label: string,
		value: string
	]>}
	{@param? label: string}
	{@param? placeholder: string}
	{@param? readOnly: bool}
	{@param? repeatable: bool}
	{@param? required: bool}
	{@param? showLabel: bool}
	{@param? tip: string}

	{call FieldBase.render}
		{param contentRenderer kind="html"}
			{if $items}
				<div class="ddm-field-options-container">
					{foreach $option in $items}
						<div class="ddm-options-target" data-index="{index($option)}"></div>

						{let $fieldConfig kind="html"}
							<div class="ddm-field-options" data-index="{index($option)}">
								<span class="ddm-options-drag {$readOnly ? 'disabled': ''}">
									<svg
										aria-hidden="true"
										class="lexicon-icon lexicon-icon-drag"
									>
										<use xlink:href="{$spritemap}#drag" />
									</svg>
								</span>

								<div class="ddm-option-entry">
									{call KeyValue.render}
										{param events: [
											'fieldBlurred': $_handleOptionBlurred,
											'fieldEdited': $_handleOptionLabelEdited,
											'fieldFocused' : $_handleOptionFocused,
											'fieldKeywordEdited': $_handleOptionValueEdited
										] /}
										{param generateKeyword: $option.generateKeyword /}
										{param keyword: $option.value /}
										{param keywordReadOnly: $defaultLanguageId != $editingLanguageId /}
										{param index: index($option) /}
										{param name: 'option' + index($option) /}
										{param placeholder: $placeholder /}
										{param readOnly: $option.disabled /}
										{param ref: 'option' + index($option) /}
										{param required: $required /}
										{param showLabel: false /}
										{param spritemap: $spritemap /}
										{param value: $option.label /}
									{/call}

									{if not isLast($option) and not $readOnly}
										<button class="close close-modal" data-onclick="{$_handleOptionDeleted}" type="button">
											<svg
												aria-hidden="true"
												class="lexicon-icon lexicon-icon-times"
											>
												<use xlink:href="{$spritemap}#times" />
											</svg>
										</button>
									{/if}
								</div>
							</div>
						{/let}

						{$fieldConfig}
					{/foreach}
				</div>
			{/if}
		{/param}

		{param id: $id /}
		{param label: $label /}
		{param repeatable: $repeatable /}
		{param showLabel: $showLabel /}
		{param spritemap: $spritemap /}
		{param tip: $tip /}
	{/call}
{/template}