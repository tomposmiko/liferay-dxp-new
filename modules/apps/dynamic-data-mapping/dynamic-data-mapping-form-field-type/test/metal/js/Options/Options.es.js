import Options from 'source/Options/Options.es';

let component;
const spritemap = 'icons.svg';

const optionsValue = {
	en_US: [
		{
			label: 'Option 1',
			value: 'Option1'
		},
		{
			label: 'Option 2',
			value: 'Option2'
		}
	]
};

describe(
	'Options',
	() => {
		beforeEach(() => jest.useFakeTimers());

		afterEach(
			() => {
				if (component) {
					component.dispose();
				}
			}
		);

		it(
			'should show the options',
			() => {
				component = new Options(
					{
						spritemap,
						value: optionsValue
					}
				);

				expect(component).toMatchSnapshot();
			}
		);

		it(
			'should delete an option when delete button is clicked',
			() => {
				component = new Options(
					{
						spritemap,
						value: optionsValue
					}
				);

				const before = component.items.length;

				component.deleteOption(0);

				expect(component.items.length).toEqual(before - 1);
			}
		);

		it(
			'.normalizeValue() should not allow options with the same value',
			() => {
				component = new Options(
					{
						spritemap
					}
				);

				const value = {
					en_US: [
						{
							label: 'One',
							value: 'One'
						},
						{
							label: 'One',
							value: 'One'
						}
					]
				};

				expect(component.normalizeValue(value, true)).toEqual(
					{
						en_US: [
							{
								label: 'One',
								value: 'One'
							},
							{
								label: 'One',
								value: 'One1'
							}
						]
					}
				);
			}
		);

		it(
			'should allow the user to order the fieldName options by dragging and dropping the options',
			() => {
				component = new Options(
					{
						spritemap,
						value: optionsValue
					}
				);

				const spy = jest.spyOn(component, 'emit');

				jest.runAllTimers();

				component.moveOption(1, 0);

				jest.runAllTimers();

				expect(component).toMatchSnapshot();

				expect(spy).toHaveBeenCalledWith('fieldEdited', expect.anything());
			}
		);
	}
);