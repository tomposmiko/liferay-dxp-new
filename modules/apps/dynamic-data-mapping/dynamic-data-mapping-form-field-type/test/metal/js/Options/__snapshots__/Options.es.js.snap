// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Options should allow the user to order the fieldName options by dragging and dropping the options 1`] = `
<div>
  <div class="form-group">
    <div class="ddm-field-options-container">
      <div class="ddm-options-target" data-index="0"></div>
      <div class="ddm-field-options" data-index="0"><span class="ddm-options-drag "><svg aria-hidden="true" class="lexicon-icon lexicon-icon-drag">
            <use xlink:href="icons.svg#drag"></use>
          </svg></span>
        <div class="ddm-option-entry">
          <div class="form-group">
            <div class="form-group"><input class="ddm-field-text form-control" name="keyValueLabeloption0" type="text" placeholder="enter-an-option" value=""></div>
            <div class="active form-text key-value-editor"><label class="control-label key-value-label">field-name:</label><input class="key-value-input" type="text" value=""></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Options should show the options 1`] = `
<div>
  <div class="form-group">
    <div class="ddm-field-options-container">
      <div class="ddm-options-target" data-index="0"></div>
      <div class="ddm-field-options" data-index="0"><span class="ddm-options-drag "><svg aria-hidden="true" class="lexicon-icon lexicon-icon-drag">
            <use xlink:href="icons.svg#drag"></use>
          </svg></span>
        <div class="ddm-option-entry">
          <div class="form-group">
            <div class="form-group"><input class="ddm-field-text form-control" name="keyValueLabeloption0" type="text" placeholder="enter-an-option" value=""></div>
            <div class="active form-text key-value-editor"><label class="control-label key-value-label">field-name:</label><input class="key-value-input" type="text" value=""></div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
